v4/newrelic/segments.go

```diff
@@ -243,11 +243,24 @@ func (s *ExternalSegment) library() string {
 	return s.Library
 }
 
+func (s *ExternalSegment) url() (*url.URL, error) {
+	if "" != s.URL {
+		return url.Parse(s.URL)
+	}
+	r := s.Request
+	if nil != s.Response && nil != s.Response.Request {
+		r = s.Response.Request
+	}
+	if r != nil {
+		return r.URL, nil
+	}
+	return nil, nil
+}
+
 func (s *ExternalSegment) host() string {
 	host := s.Host
-	if host == "" && s.URL != "" {
-		url, err := url.Parse(s.URL)
-		if err == nil {
+	if host == "" {
+		if url, _ := s.url(); url != nil {
```

Comment:
#### Bug 1
The host should be looked for in the s.Host, s.URL, and the s.Request and s.Response. We were not yet checking for it on the request/response.